RevenueCatUI/CustomerCenter/Views/PromotionalOfferView.swift

```diff
@@ -83,8 +83,14 @@ struct PromotionalOfferView: View {
                 }
             }
         }
+        .onAppear {
+            self.viewModel.onPromotionalOfferSuccessfullyPurchased = self.onPromotionalOfferSuccessfullyPurchased
+        }
     }
 
+    private func onPromotionalOfferSuccessfullyPurchased() {
+        self.dismiss()
```

Comment:
Does this close the whole Customer Center? Or only the promotional offer?

Reply:
The `dismiss` environment variable is given to us by SwiftUI, and it dismisses the current view depending on how it's presented:
- If the current view is in a NavigationStack, it pops the view off of the stack
- If the current view is presented in a sheet, it dismisses the sheet

Since the `PromotionalOfferView` is always presented in a sheet on top of other Customer Center stuff, it'll only dismiss the sheet that the PromotionalOfferView is presented in

If you're curious, you can read more about the `dismiss` environment value here: https://developer.apple.com/documentation/swiftui/environmentvalues/dismiss